### PR TITLE
Update Cluster API to v0.2.5 and update Azure dependencies

### PR DESCRIPTION
**Pull request by justaugustus**
Signed-off-by: Stephen Augustus <saugustus@vmware.com>

**What this PR does / why we need it**:
Pulls in fix for clusterctl timeouts/error handling (https://github.com/kubernetes-sigs/cluster-api/pull/1517)

/assign @vincepri @CecileRobertMichon @juan-lee
/priority important-soon
/milestone v0.3
/kind bug feature

**Release note**:

```release-note
Update Cluster API to v0.2.5 and update Azure dependencies
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -17,26 +17,24 @@ require (
...
 	k8s.io/klog v1.0.0
 	k8s.io/kube-openapi v0.0.0-20190816220812-743ec37842bf // indirect
 	k8s.io/utils v0.0.0-20190809000727-6c36bc71fc4a
-	sigs.k8s.io/cluster-api v0.2.3
-	sigs.k8s.io/controller-runtime v0.2.2
+	sigs.k8s.io/cluster-api v0.2.5
+	sigs.k8s.io/controller-runtime v0.3.0
 )
 
 replace (
```
Comment (vincepri) [inline go.mod]: You can remove these replace directives with v0.2.5

File: go.mod [hunk 2]
```diff
@@ -17,26 +17,24 @@ require (
...
 	golang.org/x/net v0.0.0-20190909003024-a7b16738d86b
-	golang.org/x/oauth2 v0.0.0-20190604053449-0f29369cfe45 // indirect
 	golang.org/x/sys v0.0.0-20190911201528-7ad0cfa0b7b5 // indirect
 	golang.org/x/text v0.3.2 // indirect
 	google.golang.org/appengine v1.5.0 // indirect
 	gopkg.in/check.v1 v1.0.0-20190902080502-41f04d3bba15 // indirect
-	k8s.io/api v0.0.0-20190831074750-7364b6bdad65
+	k8s.io/api v0.0.0-20190918195907-bd6ac527cfd2
 	k8s.io/apimachinery v0.0.0-20190831074630-461753078381
 	k8s.io/client-go v11.0.1-0.20190409021438-1a26190bd76a+incompatible
```
Comment (vincepri) [inline go.mod]: api, machinery and client-go should match CAPI

Comment (justaugustus) [reply go.mod]: One of these is doing weird/bad stuff with the Azure dependencies.
I'll take a look at that next week.

Comment (CecileRobertMichon) [reply go.mod]: I had the same issue on my branch, you have to bump autorest. https://github.com/CecileRobertMichon/cluster-api-provider-azure/blob/5ca08a7a304cb9f7641fd910ab24a01b44d019f4/go.mod should work

**Comment by CecileRobertMichon** [comment]
@justaugustus the test is complaining that `go module files are out of date`

**Comment by justaugustus** [comment]
Superseded by https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/302.
/close

**Pull request closed without merging**